scalding-core/src/main/scala/com/twitter/scalding/RichPipe.scala

```diff
@@ -52,6 +53,38 @@ object RichPipe extends java.io.Serializable {
     }
     p
   }
+
+  // A pipe can have more than one description when merged together, so we store them delimited witg 254.toChar.
+  def encodePipeDescriptions(descriptions: Seq[String]): String = {
+    descriptions.map(_.replace(254.toChar, ' ')).filter(_.nonEmpty).mkString(254.toChar.toString)
+  }
+
+  def decodePipeDescriptions(encoding: String): Seq[String] = {
+    encoding.split(254.toChar).toSeq
+  }
+
+  def getPipeDescriptions(p: Pipe): Seq[String] = {
+    if (p.getStepConfigDef.isEmpty)
+      Nil
+    else {
+      val encodedResult = p.getStepConfigDef.apply(Config.WithDescriptionSetExplicitly, new Getter {
+        override def update(s: String, s1: String): String = ???
+        override def get(s: String): String = null
+      })
+      Some(encodedResult)
+        .filterNot(x => x == null || x.isEmpty)
+        .map(decodePipeDescriptions)
+        .getOrElse(Nil)
+    }
+  }
+
+  def setPipeDescriptions(p: Pipe, descriptions: Seq[String]): Pipe = {
+    p.getStepConfigDef().setProperty(
```

Comment:
This will mostly work, but... our OrderedSerialization issues seem to have popped up that these won't always go together :/  -- we tried to use these to configure serializers for each step. Do an op that would do some transform then added the config to that pipe to let it work. But then the two would be moved apart by cascading. Not a blocker to this, but it unfortunately would mean the naming ends up being more best effort than would be ideal I think